src/csharp/Microsoft.Spark.E2ETest/Microsoft.Spark.E2ETest.csproj

```diff
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
+    <EnableUnsafeBinaryFormatterSerialization>true</EnableUnsafeBinaryFormatterSerialization>
```

Comment:
Yes, we have to, it's a [new requirement for .NET7+](https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/7.0/binaryformatter-apis-produce-errors), since binary serialization is now forbidden by default. 

![image](https://github.com/user-attachments/assets/9c0b31db-e6f3-418c-99fd-d069ea56fb56)


For .NET9, we'll have to migrate to the standalone NuGet package, since .NET8 is the last supported version.  

We can arrange a quick call where I can shed some light on background, what do we use serialization for, and alternatives to consider. 

[PR about replacing Serializer](https://github.com/dotnet/spark/pull/1166), inside there are comments on why it doesn't work

Reply:
Got it!